objects/src/accounts/package/storage_entry/word.rs

```diff
@@ -73,9 +135,17 @@ impl<'de> Deserialize<'de> for WordRepresentation {
             where
                 E: DeError,
             {
-                // Attempt to convert the input string to a Digest
+                // Attempt to deserialize as TemplateKey first
+                if let Ok(tk) = TemplateKey::try_deserialize(value) {
+                    return Ok(WordRepresentation::Dynamic(tk));
+                }
```

Comment:
Instead of trying one then the other, we could check the first two chars, if they are `0x` the we use the `parse_hex_string_as_word`. This way the error message could be different in both cases because we know which variant we were expecting.

Reply:
While implementing this I realized it is still valuable to output valid formats for string-based values. For example: maybe someone writes "123" expecting decimals to work but then it won't match with the "0x" and so we would assume a dynamic value when that's not really the case.